arch/arm/boot/dts/imx6ul-ts7100.dtsi

```diff
@@ -0,0 +1,543 @@
+	pinctrl-0 = <&pinctrl_lcd_ctrl>;
+	lcd-supply = <&reg_lcdif_enable>;
+	status = "okay";
+
+	port {
+		display_out: endpoint {
+			remote-endpoint = <&panel_in>;
+		};
+	};
+};
+
+&snvs_rtc {
+	status = "disabled";
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+&uart7 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart7>;
+	status = "okay";
+};
+
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	no-1-8-v;
+	disable-wp;
+	broken-cd = <1>;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	status = "okay";
+	fsl,ext-reset-output;
+};
+
+&weim {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_weim_fpga>;
+	clocks = <&clks IMX6UL_CLK_EIM>, <&clks IMX6UL_CLK_EIM_SLOW_SEL>;
+	ranges = <0 0 0x50000000 0x08000000>;
+	status = "okay";
+
+	#address-cells = <2>;
+	#size-cells = <1>;
+
+	fpga: fpga@50000000 {
+		compatible = "simple-bus";
+		reg = <0 0x50000000 0x00010000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0 0 0 0x10000>;
+
+		fsl,weim-cs-timing = <
+			0x0161030F	// EIM_CSnGCR1 @ 0x021b8000
+			0x00000000	// EIM_CSnGCR2 @ 0x021b8004
+			0x03000000	// EIM_CSnRCR1 @ 0x021b8008
+			0x00000000	// EIM_CSnRCR2 @ 0x021b800c
+			0x01000000	// EIM_CSnWCR1 @ 0x021b8010
+			0		// EIM_CSnWCR2 @ 0x021b8014
+		>;
+
+		clocks {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
+			fpga_clk_weim_bclk: fpga_clk_weim_bclk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+				clock-frequency = <49500000>;
+			};
+		};
+
+		syscon: syscon@4000 {
+			compatible = "simple-bus";
+			#address-cells = <0x2>;
+			#size-cells = <0x1>;
+			reg = <0x4000 0x58>;
+			ranges = <0 0 0x4000 0x58>;
+
+			gpio6: fpga_gpio6@10 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x10 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio7: fpga_gpio7@40 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x40 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio8: fpga_gpio8@50 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x50 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			fpga_intc: fpga_intc@0 {
+				compatible = "technologic,ts71xxweim-intc";
+
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				reg = <0 0x00 0x50>;
+
+				interrupt-parent = <&gpio5>;
+				interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
+			};
+		};
+
+		fpga_uart0: serial@0 {
+			compatible = "ns16550a";
+			device_type = "serial";
+			current-speed = <115200>;
+			reg-shift = <1>;
+			reg-io-width = <1>;
+			reg = <0 16>;
+			clock-frequency = <1843200>;
+			interrupt-parent = <&fpga_intc>;
+			interrupts = <0>;
+		};
+
+		opencores_spi0: spi@0 {
```

Comment:
I _think_ this should be `opencores_spi0: spi@100 {` What follows after the @ should be an address relative to the parent.